pg_search/src/postgres/customscan/pdbscan/mod.rs

```diff
@@ -330,6 +334,52 @@ impl CustomScan for PdbScan {
             let root = builder.args().root;
             let rel = builder.args().rel;
 
+            // quick look at the target list to see if we might need to do our const projections
+            let target_list = (*(*builder.args().root).parse).targetList;
+            let maybe_needs_const_projections = maybe_needs_const_projections(target_list.cast());
+
+            //
+            // look for quals we can support.  we do this first so that we can get out early if this
+            // isn't a query we can support.
+            //
+            // Opening the Directory and Index down below is expensive, so if we can avoid it,
+            // especially for non-SELECT (ie, UPDATE) statements, that's good
+            //
+            let is_select =
+                (*(*builder.args().root).parse).commandType == pg_sys::CmdType::CMD_SELECT;
+            let (quals, ri_type, restrict_info) = Self::extract_all_possible_quals(
+                &mut builder,
+                root,
+                rti,
+                restrict_info,
+                ri_type,
+                &bm25_index,
+                maybe_needs_const_projections,
+                is_select,
+            );
+
+            let Some(quals) = quals else {
+                // if we are not able to push down all of the quals, then do not propose the custom
+                // scan, as that would mean executing filtering against heap tuples (which amounts
+                // to a join, and would require more planning).
```

Comment:
Nit: this comment was already stale, but if you end up needing to iterate on this, could update it.

Reply:
If you'd like to propose a different comment I'm happy to merge it